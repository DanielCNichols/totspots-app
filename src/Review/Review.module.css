.review {
  padding: 15px 0;
  border-bottom: 2px solid #e7e7e7;
}

.review p {
  margin: 0 0 5px 0;
}

.review p:first-child {
  font-weight: bold;
}

.review p:nth-child(3) {
  font-style: italic;
  margin-bottom: 10px;
  margin-top: 10px;
}

.ratingSpan {
  font-size: 0.8em;
}
