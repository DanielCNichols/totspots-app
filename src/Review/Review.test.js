import React from 'react';
import ReactDOM from 'react-dom'
import Review from './Review.js'
import {MemoryRouter} from 'react-router-dom'
import renderer from 'react-test-renderer'

let reviews = [
  {
      "id": 8,
      "content": "Nam ultrices, libero non mattis pulvinar, nulla pede ullamcorper augue, a suscipit nulla elit ac nulla. Sed vel enim sit amet nunc viverra dapibus. Nulla suscipit ligula in lacus. Curabitur at ipsum ac tellus semper interdum. Mauris ullamcorper purus sit amet nulla. Quisque arcu libero, rutrum ac, lobortis vel, dapibus at, diam.",
      "price": 2,
      "starrating": 4,
      "volume": 5,
      "date_created": "2019-10-18T19:59:22.209Z",
      "venue_id": 1,
      "user_id": 9,
      "first_name": "Fanni",
      "last_name": "Gerriessen"
  }
]



it('renders Review without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(
  <MemoryRouter>
    <Review reviews={reviews}/>
  </MemoryRouter>, div);
  ReactDOM.unmountComponentAtNode(div)
})

it('renders Review as expected', () => {
  const tree = renderer
  .create(
  <MemoryRouter>
    <Review name="Review" reviews={reviews}/>
  </MemoryRouter>
  )
  .toJSON();
  expect(tree).toMatchSnapshot();
})
