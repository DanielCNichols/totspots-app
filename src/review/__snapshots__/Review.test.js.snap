// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Review as expected 1`] = `
<li
  className="reviewItem"
>
  <h3>
     
  </h3>
  <p>
    Visited: 
    October 23rd, 2019
  </p>
  <div
    className="rating_break"
  >
    <div
      className="userRating"
    >
      Overall: 
      <span
        className="rating"
      >
        <div>
          <p>
            No reviews
          </p>
        </div>
      </span>
    </div>
    <div
      className="userRating"
    >
      Price: 
      <span
        className="price"
      >
        <p>
          Price unavailable
        </p>
      </span>
    </div>
    <div
      className="userRating"
    >
      Volume Level: 
      <span
        className="volume"
      >
        <p
          className="NA"
        >
          volume unavailable
        </p>
      </span>
    </div>
  </div>
  <div
    className="userReview"
  >
    <p>
       said: 
    </p>
    <p />
  </div>
  <div
    className="votes"
  >
    <p>
      Was this review helpful?
    </p>
    <span>
      (VoteCount) liked this
    </span>
    <button
      onClick={[Function]}
      value="true"
    >
      Like
    </button>
  </div>
</li>
`;
